### PR TITLE
feat(components): Add configurable column width to DataTable

### PR DESCRIPTION
**Pull request by danlaurent**
## Motivations

To be able to adjust the column width manually in order to emphasize column data

### Added

- Configurable column width options in column prop (`size, minSize, maxSize`)

## Testing

- Add `size, minSize, maxSize` to a `column` object and test if this style is applied to the column element

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by kkimble006** [comment]
@danlaurent can you added definitions for these new props: Add size, minSize, maxSize to a column object and test if this style is applied to the column element. Or link to the appropriate tanstack docs?

**Comment by danlaurent** [comment]
> https://1e57cfd2.atlantis.pages.dev

@kkimble006  Yeah, sure!
You can follow [this example](https://1e57cfd2.atlantis.pages.dev/components/data-table#table-with-manual-column-sizing) that I added to the docs or the [tanstack docs](https://tanstack.com/table/v8/docs/api/features/column-sizing?from=reactTableV7&original=https://react-table-v7.tanstack.com/docs/api/useResizeColumns#column-def-options).
After that you can inspect the element and see if the styling properties were applied

**Comment by danlaurent** [comment]
> @danlaurent I did have a question about the prop names themselves - given that this specifically controls column width, did you consider `width`, `minWidth` and `maxWidth`? Those would map more cleanly to the CSS underneath than `size` which I feel could lend itself to assumptions about having control over both height and width

@chris-at-jobber Yeah, I totally agree, I would prefer to use `width`, but as we're using the built in Column Sizing API from Tanstack, this is the nomenclature that the library is using, we're only consuming it. Here's the [docs](https://tanstack.com/table/v8/docs/api/features/column-sizing?from=reactTableV7&original=https://react-table-v7.tanstack.com/docs/api/useResizeColumns#size)

**Comment by chris-at-jobber** [comment]
Oh so do we have any control over the props interface with DataTable? Or are we totally tied to TanStack's naming?

**Comment by danlaurent** [comment]
> Oh so do we have any control over the props interface with DataTable? Or are we totally tied to TanStack's naming?

@chris-at-jobber I mean, we could add 3 extra "custom" props to this interface (by extending the library type) but I have some concerns:
- All other helper methods are using the same nomenclature (`getSize`, `startSize`)
- I didn't take a look on how the library built the helper methods, but by using custom props in the columnDef maybe those methods won't work properly

**Comment by chris-at-jobber** [comment]
Ahhhhhh OK I see what you're saying. Thanks for the clarification!

File: packages/components/src/DataTable/DataTable.mdx [hunk 1]
```diff
@@ -662,6 +662,7 @@ import { DataTable } from "@jobber/components/DataTable";
         accessorKey: "name",
         header: "Name",
         footer: "Totals",
+        enableResizing: false,
```
Comment (MichaelParadis) [inline packages/components/src/DataTable/DataTable.mdx:665]: Is this needed?

**Review by MichaelParadis — APPROVED** [review]
LGTM! Minor comment but looks good

File: packages/components/src/DataTable/DataTable.mdx (hunk 1 above)
Comment (danlaurent) [reply packages/components/src/DataTable/DataTable.mdx:665]: Not needed, I'll remove it, thanks!

**Pull request merged**